src/tools/auth0/handlers/scimHandler.ts

```diff
@@ -0,0 +1,228 @@
+			const headers = { 
+				'Accept': 'application/json',
+				'Authorization': `Bearer ${ accessToken }`
+			}
+			options = [...options, { headers }];
+
+			return await axios[method](...options);
+		});
+	}
+
+	/**
+	 * Error handler wrapper.
+	 */
+	async withErrorHandling(callback) {
+		try {
+			return await callback();
+		} catch (error) {
+			const errorData = error?.response?.data;
+			if (errorData?.statusCode === 404) throw "SCIM_NOT_FOUND";
+
+			const statusCode = errorData?.statusCode || error?.response?.status;
+			const errorCode = errorData?.errorCode || errorData?.error || error?.response?.statusText;
+			const errorMessage = errorData?.message || error?.response?.statusText;
+			const message = `SCIM request failed with statusCode ${ statusCode } (${ errorCode }). ${ errorMessage }.`;
+
+			log.error(message);
+			throw error;
+		}
+	}
+
+	/**
+	 * Returns formatted endpoint url.
+	 */
+	private getScimEndpoint(connection_id: string) {
+		// Call `scim-configuration` endpoint directly to support `SCIM` features.
+		return `https://${ this.config('AUTH0_DOMAIN') }/api/v2/connections/${ connection_id }/scim-configuration`;
+	}
+
+	/**
+	 * Creates a new `SCIM` configuration.
+	 */
+	async createScimConfiguration({ id: connection_id }: scimRequestParams, { user_id_attribute, mapping }: scimBodyParams): Promise<AxiosResponse> {
+		log.debug(`Creating SCIM configuration for connection ${ connection_id }`);
+		const url = this.getScimEndpoint(connection_id);
+		return (await this.scimHttpRequest('post', [ url, { user_id_attribute, mapping } ])).data;
+	}
+
+	/**
+	 * Retrieves `SCIM` configuration of an enterprise connection.
+	 */
+	async getScimConfiguration({ id: connection_id }: scimRequestParams): Promise<scimBodyParams> {
+		log.debug(`Getting SCIM configuration from connection ${ connection_id }`);
+		const url = this.getScimEndpoint(connection_id);
+		return (await this.scimHttpRequest('get', [ url ])).data;
+	}
+
+	/**
+	 * Updates an existing `SCIM` configuration.
+	 */
+	async updateScimConfiguration({ id: connection_id }: scimRequestParams, { user_id_attribute, mapping }: scimBodyParams): Promise<AxiosResponse> {
+		log.debug(`Getting SCIM configuration on connection ${ connection_id }`);
+		const url = this.getScimEndpoint(connection_id);
+		return (await this.scimHttpRequest('patch', [ url, { user_id_attribute, mapping } ])).data;
+	}
+
+	/**
+	 * Deletes an existing `SCIM` configuration.
+	 */
+	async deleteScimConfiguration({ id: connection_id }: scimRequestParams): Promise<AxiosResponse>  {
+		log.debug(`Getting SCIM configuration of connection ${ connection_id }`);
+		const url = this.getScimEndpoint(connection_id);
+		return (await this.scimHttpRequest('delete', [ url ])).data;
+	}
+  
+	async updateOverride(requestParams: scimRequestParams, bodyParams: Asset) {
+		// Extract `scim_configuration` from `bodyParams`.
+		// Remove `scim_configuration` from `bodyParams`, because `connections.update` doesn't accept it.
```

Comment:
We can add debug logs to this and few other newly added methods.

Reply:
`updateScimConfiguration`, `createScimConfiguration`, and `deleteScimConfiguration` are called within `updateOverride`, which already has a debug logger in place.